src/components/elements/table/RestockTableMain.tsx

```diff
@@ -0,0 +1,214 @@
+import React, {useEffect} from 'react';
+
+import Paper from '@material-ui/core/Paper';
+import {createStyles, makeStyles, Theme} from '@material-ui/core/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TablePagination from '@material-ui/core/TablePagination';
+import TableRow from '@material-ui/core/TableRow';
+import TableSortLabel from '@material-ui/core/TableSortLabel';
+
+import {RestockDataInfo} from '../../../state/restock/restockData';
+import {restockDispatchers} from '../../../state/restock/restockDispatchers';
+import {useRestockSelector} from '../../../state/restock/restockSelector';
+import {useDispatch} from '../../../state/store';
+import {stableSort, getComparator, Order} from '../../../utils/Sort';
+import PurchaseModal from './PurchaseModal';
+import Status from './Status';
+
+const statusColors : { [key: string]: any } = {
+  completed: 'success',
+  processing: 'info',
+  shipped: 'danger',
+};
+
+type HeadCell = {
+  id: keyof RestockDataInfo;
+  label: string;
+  numeric: boolean;
+}
+
+const headCells: HeadCell[] = [
+  {id: 'id', numeric: false, label: 'Order #'},
+  {id: 'purchaseDate', numeric: false, label: 'Purchase Date'},
+  {id: 'status', numeric: false, label: 'Status'},
+  {id: 'type', numeric: false, label: 'Type'},
+  {id: 'totalPrice', numeric: true, label: 'Total Price ($)'},
+];
+
+type EnhancedTableProps = {
+  classes: ReturnType<typeof useStyles>;
+  onRequestSort: (event: React.MouseEvent<unknown>, property: keyof RestockDataInfo) => void;
+  order: Order;
+  orderBy: string;
```

Comment:
`orderKey` may sound better (the key used for sorting) or maybe make this a type of function `(item) => any` , the return is the thing used for comparison.

For example, sorting an object which has a key `value` will be `(item) => item.value`